dream_layer_backend/shared_utils.py

```
@@ -286,4 +287,211 @@ def upload_controlnet_image(file, unit_index: int = 0) -> Dict[str, Any]:
+                "status": "error",
+                "message": "No file provided or no file selected"
+            }, 400
+
+        # Validate file extension - only .safetensors for now
+        allowed_extensions = {'.safetensors'}
+        file_ext = Path(file.filename).suffix.lower()
+
+        if file_ext not in allowed_extensions:
+            return {
+                "status": "error",
+                "message": f"Invalid file type. Only .safetensors files are supported"
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)